sqlmesh/core/context.py

```diff
@@ -2148,6 +2157,36 @@ def _load_requirements(self, path: Path) -> None:
                         )
                     self._requirements[dep] = ver
 
+    def _load_multiple_gateways(self) -> None:
+        # Validate there are multiple gateways and the default is duckdb
+        if (
+            len(self.config.gateways) < 2
+            or not isinstance(self._connection_config, DuckDBConnectionConfig)
+            or not self._connection_config.database
+            or self._connection_config.catalogs
+        ):
+            return
+
+        # Set the attach options for the main and additional connection gateways (Currently supported DuckDB and PostgreSQL)
+        self._connection_config.catalogs = {
+            self._connection_config.database.split(".")[0]: self._connection_config.database
+        }
+        for gateway_name, config in self.config.gateways.items():
+            if isinstance(config.connection, PostgresConnectionConfig):
+                self._connection_config.catalogs[gateway_name] = DuckDBAttachOptions(
+                    type=config.connection.type_, path=config.connection._libpq_string
+                )
```

Comment:
I'm not sure about mutating the connections' `catalogs` attribute here.. Curious what others think, but my hunch is we may wanna push this down to the config validators or something.

Reply:
Good idea, this makes more sense I'll work on moving it there